### PR TITLE
SpinButton: updated base resetstyle for input of spin button

### LINKED ISSUE #27398: [Bug]: SpinButton increment and decrement button do not adjust to size of SpinButton
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 5.15 Linux Mint 20.3 (Una)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 16.02 GB / 31.02 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Browsers:
    Chrome: 110.0.5481.177
    Chromium: 95.0.4638.54
    Firefox: 110.0
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/damp-hill-i0u22p?file=/src/App.tsx

## Actual Behavior
If the size of the SpinButton is changed, the increment and decrement buttons do not change position.
image

## Expected Behavior
The increment and decrement buttons should appear inside the input element.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by kkakroo**
## Previous Behavior
Inc and Dec buttons were not respecting the change width to the spin button 

## New Behavior
Inc and Dec buttons are now respecting the change of width to the spin button as the input styles have been updated.

## Related Issue(s)

- Fixes #27398

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                Change |
| :--------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>           | `35.648 kB`<br />`10.348 kB` | `35.665 kB`<br />`10.353 kB` | `17 B` <br />`5 B`  |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButtonField.fixture.js'>SpinButtonField</abbr> | `44.671 kB`<br />`12.812 kB` | `44.688 kB`<br />`12.814 kB` | `17 B` <br />`2 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.994 kB`<br />`18.395 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `207.512 kB`<br />`57.555 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `38.819 kB`<br />`12.351 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.446 kB`<br />`2.185 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/cb6569b754d05385466021717a283bdead3f3f34'>cb6569b754d05385466021717a283bdead3f3f34</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Avatar-mount_base.html">907</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Avatar-mount.html">895</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Button-mount_base.html">589</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Button-mount.html">590</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Field-mount_base.html">1521</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Field-mount.html">1533</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProvider-mount_base.html">1093</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProvider-mount.html">1090</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProviderWithTheme-mount_base.html">295</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProviderWithTheme-mount.html">293</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">278</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">284</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">285</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/InfoButton-mount_base.html">204</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/InfoButton-mount.html">204</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/MakeStyles-mount_base.html">1393</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/MakeStyles-mount.html">1398</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Persona-mount_base.html">2129</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/Persona-mount.html">2045</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/SpinButton-mount_base.html">1871</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27421/perf-test-react-components/SpinButton-mount.html">1823</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts [hunk 1]
```diff
@@ -223,6 +223,7 @@ const useInputClassName = makeResetStyles({
   fontSize: 'inherit',
   fontWeight: 'inherit',
   lineHeight: 'inherit',
+  width: '100%',
```
[the remark this answers is not in the document]
Comment (kkakroo) [reply packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts:226]: sure @spmonahan, let me give that a shot.

[the remark this answers is not in the document]
Comment (kkakroo) [reply packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts:226]: Hi @spmonahan, I have added a VR test for the custom width scenario.

File: apps/vr-tests-react-components/src/stories/SpinButton.stories.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { storiesOf } from '@storybook/react';
 import { SpinButton, spinButtonClassNames } from '@fluentui/react-spinbutton';
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/SpinButton.stories.tsx:4]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 4 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27421&commitId=844f16ea06cc38f1be0d1eae31802dc8f78d766e&env=prod)

✅ There was 4 screenshots added, 0 screenshots removed, 1926 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SpinButton Converged.Custom Width - Dark Mode.chromium.png|0| Added|
|SpinButton Converged.Custom Width - High Contrast.chromium.png|0| Added|
|SpinButton Converged.Custom Width - RTL.chromium.png|0| Added|
|SpinButton Converged.Custom Width.chromium.png|0| Added|

</details>


<div id="vrtCommentfluentuiv9"/>

Comment (spmonahan) [reply apps/vr-tests-react-components/src/stories/SpinButton.stories.tsx:4]: Looks good to me!

File: packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts:226]: Thanks!

**Review by spmonahan — APPROVED** [review]

**Pull request merged**